test/unit/io/sequence_file/sequence_file_output_test.cpp

```diff
@@ -202,6 +196,39 @@ TEST(general, default_template_args_and_deduction_guides)
     }
 }
 
+TEST(general, valid_file_extensions)
+{
+    {  // Single format
+        sequence_file_output fout{std::ostringstream{}, sequence_file_format_fasta{}};
+
+        EXPECT_TRUE(std::ranges::equal(decltype(fout)::valid_file_extensions(),
+                                       sequence_file_format_fasta::file_extensions));
+    }
+
+    {  // All formats
+        // get all extensions.
+        auto all_extensions = sequence_file_output<>::valid_file_extensions();
+
+        // define testing lambda
+        auto cmp_lambda = [&all_extensions](auto & source)
+        {
+            return std::find(all_extensions.begin(), all_extensions.end(), source);
+        };
+
+        // Test fasta extensions
+        for (std::string ext : sequence_file_format_fasta::file_extensions)
```

Comment:
```suggestion
        for (std::string & ext : sequence_file_format_fasta::file_extensions)
```
also for all the other ones